src/AggregateFunctions/SingleValueData.h

```diff
@@ -376,6 +375,36 @@ struct SingleValueDataGenericWithColumn final : public SingleValueDataBase
 
 static_assert(sizeof(SingleValueDataGenericWithColumn) <= SingleValueDataBase::MAX_STORAGE_SIZE, "Incorrect size of SingleValueDataGenericWithColumn struct");
 
+/// A variant of SingleValueData that does not own the value, but rather references a value in a column
+struct SingleValueReference final : public SingleValueDataBase
+{
+    static constexpr bool is_compilable = false;
+    using Self = SingleValueReference;
+
+    ColumnPtr column_ref;
```

Comment:
Maybe it should be a ref indeed, I mean `IColumn&`. We shouldn't hold the last reference to the column here, right?